### PR TITLE
TensorFlow static extensions, SDCA multiclass static extensions

### PR DESCRIPTION
**Pull request by Zruty0**
Code changes to enable end-to-end static pipe training for TF and multiclass

### CODE REVIEW

File: src/Microsoft.ML.Data/StaticPipe/TrainerEstimatorReconciler.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Internal.Utilities;
+using static Microsoft.ML.Data.StaticPipe.Runtime.TrainerEstimatorReconciler;
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/StaticPipe/TrainerEstimatorReconciler.cs]: >Microsoft.ML.Data.StaticPipe.Runtime.TrainerEstimatorReconciler [](start = 13, length = 63)

Hmmm. What's this for? Do we need it?

File: src/Microsoft.ML.Data/StaticPipe/TrainerEstimatorReconciler.cs [hunk 2]
```diff
@@ -336,5 +337,71 @@ private sealed class ImplBool : Scalar<bool>
...
+
+            protected override IEnumerable<PipelineColumn> Outputs => new PipelineColumn[] { Output.score, Output.predictedLabel };
+
+            private static readonly string[] _fixedOutputNames = new[] { DefaultColumnNames.Score, DefaultColumnNames.PredictedLabel };
+
+            /// <summary>
+            /// Constructs a new general regression reconciler.
+            /// </summary>
+            /// <param name="estimatorFactory">The delegate to create the training estimator. It is assumed that this estimator
+            /// will produce a single new scalar <see cref="float"/> column named <see cref="DefaultColumnNames.Score"/>.</param>
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/StaticPipe/TrainerEstimatorReconciler.cs]: >single new scalar <see cref="float"/> column named <see cref="DefaultColumnNames.Score"/> [](start = 31, length = 89)

The estimator I believe will produce two things, a vector float named `Score` and a key valued column named `PredictedLabel`, I think.

File: test/Microsoft.ML.Tests/TensorFlowEstimatorTests.cs [hunk 3]
```diff
@@ -137,6 +138,50 @@ void TestCommandLine()
...
+
+            using (var env = new TlcEnvironment())
+            {
+                var imageHeight = 32;
+                var imageWidth = 32;
+                var dataFile = GetDataPath("images/images.tsv");
+                var imageFolder = Path.GetDirectoryName(dataFile);
+
+                var data = TextLoader.CreateReader(env, ctx => (
+                    ImagePath: ctx.LoadText(0),
```
Comment (TomFinley) [inline test/Microsoft.ML.Tests/TensorFlowEstimatorTests.cs]: >ImagePath [](start = 20, length = 9)

Remember the discussion about the casing of these names.

File: src/Microsoft.ML.Api/PredictionFunction.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+        public TDst Predict(TSrc example) => _engine.Predict(example);
+    }
+
+    public static class PredictionFunctionExtensions
+    {
+        public static PredictionFunction<TSrc, TDst> MakePredictionFunction<TSrc, TDst>(this ITransformer transformer, IHostEnvironment env)
+                where TSrc : class
+                where TDst : class, new()
+            => new PredictionFunction<TSrc, TDst>(env, transformer);
```
Comment (TomFinley) [inline src/Microsoft.ML.Api/PredictionFunction.cs]: I guess you're in a hurry, but certainly this thing bears some explanation.

File: src/Microsoft.ML.Api/PredictionFunction.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime.Api;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    public sealed class PredictionFunction<TSrc, TDst>
```
Comment (sfilipi) [inline src/Microsoft.ML.Api/PredictionFunction.cs]: >public [](start = 4, length = 6)

XML comments on anything public

**Pull request merged**